【課題提出】其の三 『WEBページの基本的なレイアウト1』安藤

andoazumi opened the pull request:
@tommyKazuhito
@lineatakano

課題其の三 『WEBページの基本的なレイアウト1』の作成を行いましたので、ご確認よろしくお願いいたします。
制作時間8時間

「BANNER」のところが未完成です。縦横比を保ち可変する正方形を作れませんでした。cssの80行目あたりのコメントアウト部分は、ネットで調べてまわって持ってきてみましたがうまく作用しなかったものです。文字を含んで作ると文字が正方形の外に追い出され、その分の高さもプラスとなって正方形じゃなくなりflexを使った中央揃えが効かなくなっているようです。beforeの使い方をどうにかしたらできるでしょうか。

なんとか形にはできたと思ってますが、「MAIN VISUAL」をウィンドウに合わせつつ長さが違う「SIDEBAR」と並ばせるのをどうやって作っていったらいいのかとても悩みました。片方だけ可変のをやろうとしてヘッダー＆メインビジュアルとメインビジュアル＆サイドバーで括るの両立させる方法をしばらく探してました。
もっとシンプルにできる方法があれば知りたいです。

あと、訓練校時代は一気にhtmlを作ってまとめてcssをつけていてたのですが、高野さんに「項目ごとに作った方がいいよ」とアドバイスいただいて、そうやって作った方がわかりやすかったのと私の複雑化させる癖も抑えられてよかったです。ありがとうございます！

andoazumi committed:
[Add]其の三 『WEBページの基本的なレイアウト1』の作成

未完。BANNERの縦横比を保ち可変する正方形を作れませんでした。cssの80行目あたりのコメントアウト部分は、ネットで調べて持ってきてみたけれどうまく作用しなかったものです。

tommyKazuhito commented on basic-layout/no3/htdocs/css/style.css:
```diff
@@ -14,3 +30,267 @@ body {
+    margin-right: 30px;
+  }
+
+  .contents_1 .c1-items:not(:last-child) {
+    margin-bottom: none;
+  }
```
@andoazumi 
これ効いてます？？

tommyKazuhito commented on basic-layout/no3/htdocs/css/style.css:
```diff
@@ -14,3 +30,267 @@ body {
+
+  .contents_2 .inner {
+    display: grid;
+    grid-template-rows: 235px 235px 235px 235px;
```
@andoazumi 
grid-template-xxは繰り返しが使えるので、同じサイズの反復なら
`grid-template-rows: repeat(4, 235px);`
みたいな書き方ができます！

tommyKazuhito commented on basic-layout/no3/htdocs/css/style.css:
```diff
@@ -14,3 +30,267 @@ body {
+  .contents_2 .inner {
+    display: grid;
+    grid-template-rows: 235px 235px 235px 235px;
+    grid-gap: 30px 0;
```
@andoazumi 
`gap`だけで今は大丈夫です！ちなみにFlexにも変えます。
`grid-gap`ってちょっと古い書き方かな。

tommyKazuhito commented on basic-layout/no3/htdocs/index.html:
```diff
@@ -15,7 +17,63 @@
+          </div>
+        </div>
+
+        <section class="contents_1">
+          <div class="inner">
+            <div class="c1-items">
+              <p class="txt">CONTENTS 1 - 1</p>
+            </div>
+            <div class="c1-items">
+              <p class="txt">CONTENTS 1 - 2</p>
+            </div>
+          </div>
+        </section>
```
@andoazumi 
**HTMLの文法について**
`section` タグは `h2 - h6` 要素を内包していないといけません。
基礎的な文法は、ちょっと勉強してみてください。

クラスの命名規則も、ハイフンとアンスコの使い分けの意図とかも気にしたほうがいいかも。

あと、これも癖付けなんですが、HTMLをマークアップしたら、
HTMLのバリデータで自分のHTMLに問題がないか、確認するようにしましょう。
これは、みんな実際の案件でもやっています。
[The W3C Markup Validation Service](https://validator.w3.org/#validate_by_input)

tommyKazuhito commented on basic-layout/no3/htdocs/index.html:
```diff
@@ -15,7 +17,63 @@
+          </div>
+        </section>
+
+        <section class="banner">
+          <div class="inner">
+            <p class="txt img">BANNER<img src="" alt=""></p>
+          </div>
+        </section>
```
@andoazumi 
デザインの仕様を確認してみてください。
ここはコンテンツ全体、画像にしてOKです。

ただ、ここ苦労したみたいなので、CSSでいく方法のヒントを。
今、正方形のBOXまでは作れているみたいなので、
文字を `span` とかで囲ってあげて `position` で中央配置してみましょう。

tommyKazuhito reviewed (commented):
@andoazumi 
見させてもらいました！
苦労の跡がソースから見えますね。
いくつかコメントつけましたので、ご確認お願いします！

MAIN VISUALのやり方は、CSSだけでやるなら安藤さんがやってくれた方法がベストだと思います！

あと、添付の部分、SPでグレーが切れちゃってますね！
![sp](https://user-images.githubusercontent.com/22440274/191777050-cc2e9c92-2e28-4a69-8372-950f5d8560e6.png)

andoazumi commented:
@tommyKazuhito 
ご確認いただき、ありがとうございます！
私が詰んでたところへのアドバイスと、公の場でやらかす前に直した方がいい点を細かく見つけていただきありがとうございます。文法ミスのところ本当に知らなかったのでかなり危険だなって思いました。文法確認いたします。

ご指摘いただいたところを修正して再提出いたします。よろしくお願いいたします。

lineatakano commented:
@andoazumi 
確認させていただきました。
まだ、途中とのことだったので、これから作業かもしれませんが、
ウィンドウサイズをタブレット手前の大きさにした時に、
サイドバーの幅が狭くなります。

contents-2-*の文字の幅の影響だと思うのですが、何かしら対応してもらった方がよいかもですね。

lineatakano commented:
@andoazumi
確認させていただきました。
まだ、途中とのことだったので、これから作業かもしれませんが、
ウィンドウサイズをタブレット手前の大きさにした時に、
サイドバーの幅が狭くなります。

contents-2-*の文字の幅の影響だと思うのですが、何かしら対応してもらった方がよいかもですね。

andoazumi commented:
@lineatakano 
高野さん！ご確認いただき、ありがとうございます。
気づきませんでした……。確認して修正いたします。
ありがとうございます！

andoazumi committed:
[Update]課題1_3訂正

【html】
全体
sectionタグの文法に則るため、pタグをh1タグに変更。
BANNERは画像に差し替えたため、divタグに変更。

69行目
正方形作成用。

【css】
BANNER画像用のcssを追加。

73行目
正方形作成用。

225~227行目
.contents_1 .c1-items:not(:last-child) {
    margin-bottom: none;
  }
メディアクエリが(min-width:768px)であり不必要だったため、上記を削除。
書くのであれば、正しくは下記の通りでした。文法ミスです。
.contents_1 .c1-items:not(:last-child) {
    margin-bottom: 0;
  }

235行目
上記から下記に変更
grid-template-rows: 235px 235px 235px 235px;
↓
grid-template-rows: repeat(4, 235px);

280行目
上記から下記に変更
grid-template-columns: auto auto auto auto;
↓
grid-template-columns: repeat(4, auto);

168,236,280行目
「grid-gap」を「gap」に変更

254行目
.innerを追加して灰色が全体にかかるように調整しました。
 .banner .inner {
    margin-bottom: 75px;
  }

andoazumi commented:
@tommyKazuhito 
@lineatakano 

課題1_3の再提出を行いました。
コミットする際のコメントに書き忘れてしまったBANNER画像用のcssを追加した部分は、
166~172行目、243~249行目、298~304行目になります。

高野さんにご指摘いただいた部分ですが、修正できておりません。
mainタグで括っているところに圧迫されていると思い探したのですが、どこが制御できてないのかわかりませんでした。
ヒントを再度いただけたら嬉しいです。よろしくお願いいたします。

andoazumi commented:
再提出追記です。

正方形の上に文字を持ってこれたのですが、beforeの仕組みを理解しきれず中央からズレております。
beforeの確認いたします。

tommyKazuhito commented on basic-layout/no3/htdocs/index.html:
```diff
@@ -21,57 +21,66 @@
+        <div class="banner">
           <div class="inner">
-            <p class="txt img">BANNER<img src="" alt=""></p>
+            <p class="img img_pc"><img src="./img/BANNER_pc.jpg" alt="BANNER"></p>
+            <p class="img img_other"><img src="./img/BANNER_other.jpg" alt="BANNER"></p>
```
@andoazumi 
画像が表示されていないみたいです…

ファイル・ディレクトリ名は、そういう指示がない限りは基本的に半角英数小文字にしましょう。
あと、デバイスごとに画像分けるときは、 `picture` + `source` を使ってあげるといいです！

tommyKazuhito commented on basic-layout/no3/htdocs/css/style.css:
```diff
@@ -277,15 +286,23 @@ body {
+    grid-template-columns: repeat(4, auto);
     grid-template-rows: 400px;
-    grid-gap: 0 50px;
+    gap: 0 50px;
```
@andoazumi 
LGTM!

tommyKazuhito reviewed (commented):
@andoazumi 
確認しました！
HTMLのバリデーションってちゃんとかけました？
警告出てますよ？？

タブレットサイズ手前で崩れちゃうのは、
文字の折返しが原因です！
文字の折返しに関するCSS調べてみましょう！

lineatakano commented:
@andoazumi 
返事が遅れました。
tommyさんの指摘通り文字の折り返しが原因です。
また、今回は問題ないですが、flexやgridの子要素の幅がautoや指定なしなどだと、
レイアウトが想定している通りにならない可能性がありますね。

andoazumi commented:
@tommyKazuhito 
ご確認いただきありがとうございます。
再提出する前に再度チェックしてもらっていませんでした。提出する前に行うこと癖をつけます。

もう少し折り返しあたりのcssを調べてみます。
ありがとうございます。

andoazumi commented:
@lineatakano 
ご確認いただき、ありがとうございます。
折り返しのあたりで何がいけないのか調べます。
そういうことに起こってしまうのですね！気をつけます。ありがとうございます。

andoazumi committed:
[Update]課題1_3訂正

【html】
全体
sectionタグの文法に則るため、contents1とcontents2をdivタグに変更。

【css】
53行目
.main{
 	flex: 1 1 auto; }
サイドバーの横幅が動かないようにするために上記から下記に変更
 .main {
 	flex: 1;
}

63行目
sp(155行目)とtab(234行目)で別々に入れていたので一つにまとめました。
.contents_2 .inner {
  display: grid;
}

218行目
幅を50%に変更しました。

236行目
折り返しするようにword-break: break-all;を入れました。

273行目
contents1とcontents2をdivタグにしたので変更しました。

andoazumi commented:
@tommyKazuhito 
@lineatakano 

お疲れ様です。
課題1_3の再々提出をいたしましたので、お手隙の際にご確認の程よろしくお願いいたします。

tommyKazuhito reviewed (commented):
@andoazumi 

修正確認しました！
下記2点ほどご確認・ご対応をお願いいたします！

- SPサイズでコンテンツエリアの左右の余白がなくなってしまっています。
- CONTENTS2だけ、CONTENTSと2の間の半角スペースがなくなってしまっています。

細かいところですが、見てみてくださいm(_ _)m

lineatakano commented on basic-layout/no3/htdocs/css/style.css:
```diff
@@ -14,3 +30,284 @@ body {
+  }
+
+  .contents_2 .inner {
+    grid-template-columns: repeat(4, auto);
```
@andoazumi
MGTでも話がでたので、
repeat(4, auto)ではなく、repeat(4, 1fr)を試してみるのもいいかもですね。

lineatakano commented on basic-layout/no3/htdocs/css/style.css:
```diff
@@ -14,3 +30,284 @@ body {
+  }
+
+  .contents_2 .inner {
+    grid-template-columns: repeat(4, auto);
```
@andoazumi
MGTでも話がでたので、
repeat(4, auto)ではなく、repeat(4, 1fr)を試してみるのもいいかもですね。

andoazumi replied:
@lineatakano 
ご確認いただき、ありがとうございます。
autoとの違いを学ぶべく使ってみます。

andoazumi commented:
@tommyKazuhito 
ご確認いただき、ありがとうございます。
確認不足ですみません。確認して修正いたします。

andoazumi committed:
[Update]課題1_3訂正

【html】
contents2
CONZTENTSと数字の間にあるハイフンを追記しました。
数字とハイフンをまとめて折り返しさせるものを復活させました。

【css】
42行目
contents2の折り返しのcssを復活させました。

132行目
contents群の変更に伴い対応するように修正いたしました。

156行目
grid-templateで同じ内容が続くため、repeatにしました。

279行目
なかったので追記しました。

289行目
autoを1frにしました。

andoazumi commented:
@tommyKazuhito 
@lineatakano 

お疲れさまです。
修正して再提出いたしましたので、お手隙の際にご確認の程よろしくお願い申し上げます。

tommyKazuhito reviewed (approved):
LGTM!
